src/main/java/com/gmail/val59000mc/maploader/CaveOresOnlyPopulator.java

```diff
@@ -23,33 +37,38 @@
             BlockFace.WEST
     };
 
-    private static final Material AIR;
-    private static final Material CAVE_AIR;
-
-    static{
-        AIR = Material.AIR;
-        CAVE_AIR = UniversalMaterial.CAVE_AIR.getType();
-    }
-
     @Override
     public void populate(World world, Random random, Chunk chunk){
         scanChunk(chunk);
     }
 
     private void scanChunk(Chunk chunk){
+    	
+    	
+    	
+    	for (int x = 0; x < 16; x++)
+            for (int y = 5; y < 128; y++)
+                for (int z = 0; z < 16; z++)
+                	explored[x][y][z] = false;
+    	
         for (int x = 0; x < 16; x++){
-            for (int y = 5; y < 30; y++){
+            for (int y = 5; y < 128; y++){
                 for (int z = 0; z < 16; z++){
 
                     Block block = chunk.getBlock(x, y, z);
+                    
                     Material type = block.getType();
                     if (
                             type == Material.DIAMOND_ORE ||
                             type == Material.GOLD_ORE ||
-                            type == Material.LAPIS_ORE
+                            type == Material.LAPIS_ORE ||
+                            type == Material.COAL_ORE ||
+                            type == Material.IRON_ORE ||
+                            type == Material.REDSTONE_ORE ||
+                            type == Material.EMERALD_ORE
```

Comment:
Why did you decide to add these ores? I intentionally left them out to speed up the process. This is also why I left the max checking height at 30.

Reply:
Because the process is now fast enough to handle all these blocks up to 128 height